pkg/actions/kv_run_results_iterator.go

```diff
@@ -3,71 +3,77 @@ package actions
 import (
 	"context"
 
-	"google.golang.org/protobuf/proto"
-
-	"github.com/treeverse/lakefs/pkg/logging"
-
 	"github.com/treeverse/lakefs/pkg/kv"
 )
 
 type KVRunResultIterator struct {
 	it  kv.MessageIterator
-	ctx context.Context
+	err error
 }
 
-func NewKVRunResultIterator(ctx context.Context, store kv.StoreMessage, repositoryID, branchID, commitID, prefix string) *KVRunResultIterator {
-	var key string
-	var err error
+func NewKVRunResultIterator(ctx context.Context, store kv.StoreMessage, repositoryID, branchID, commitID, after string) (*KVRunResultIterator, error) {
```

Comment:
Are we allowing a mix branch + commitID ? Maybe we should assert on that?

Reply:
Current behavior in DB service is to first check branch and then commit - if branch and commit are provided only branch is considered, we do not error in that case.
I'm not sure we want to change this behavior - maybe we should do the validation on the controller side?

Reply:
Still - I think that a function accepting arguments should handle them, or assert if the input isn't valid (like a branch & commit combo).

Reply:
I can change that - but we need to understand this is a change in behavior

Reply:
The new verification will do for now.